pkg/cmd/pulumi/package_add.go

```diff
@@ -270,12 +317,34 @@ func printGoLinkInstructions(root string, pkg *schema.Package, out string) error
 		return errors.New("failed to import go language info")
 	}
 
-	fmt.Printf("   go mod edit -replace %s=%s\n", goInfo.ModulePath, relOut)
-	fmt.Println()
-	fmt.Println("You can then use the SDK in your Go code with:")
-	fmt.Println()
-	fmt.Printf("  import \"%s\"\n", goInfo.ModulePath)
-	fmt.Println()
+	gomodFilepath := filepath.Join(root, "go.mod")
+	gomodFileContent, err := os.ReadFile(gomodFilepath)
+	if err != nil {
+		return fmt.Errorf("cannot read mod file: %w", err)
+	}
+
+	gomod, err := modfile.Parse("go.mod", gomodFileContent, nil)
+	if err != nil {
+		return fmt.Errorf("mod parse: %w", err)
+	}
+
+	err = gomod.AddReplace(goInfo.ImportBasePath, "", relOut, "")
```

Comment:
This looks wrong to me. ImportBasePath is the path to the _package_ but we need the path to the _module_ here. `ModulePath` would be that if it was filled in, but if its empty I think we need to copy the default code for it from the go codegen.